pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator.go

```diff
@@ -97,6 +111,30 @@ func (mutator *VMsMutator) Mutate(ar *admissionv1.AdmissionReview) *admissionv1.
 	}
 }
 
+func (mutator *VMsMutator) applyFlavorToVm(vm *v1.VirtualMachine, ar *admissionv1.AdmissionReview) error {
+	if ar.Request.Operation == admissionv1.Create {
+		flavorProfile, err := mutator.FlavorMethods.FindProfile(vm)
+		if err != nil || flavorProfile == nil {
+			return nil
+		}
+
+		mutator.FlavorMethods.ApplyToVmi(k8sfield.NewPath("spec", "template", "spec"), flavorProfile, &vm.Spec.Template.Spec)
+	} else if ar.Request.Operation == admissionv1.Update {
+
+		oldVM := v1.VirtualMachine{}
+
+		err := json.Unmarshal(ar.Request.OldObject.Raw, &oldVM)
+		if err != nil {
+			return err
+		}
+		//in case user would like to change flavor, revert it back to original state
+		if !reflect.DeepEqual(oldVM.Spec.Flavor, vm.Spec.Flavor) {
+			vm.Spec.Flavor = oldVM.Spec.Flavor
+		}
```

Comment:
unless the flavor is being removed from the VM, this should cause a validation error in the validation webhook instead of re-applying the flavor to the VM.Spec